src/VecSim/spaces/L2_space.cpp

```diff
@@ -16,9 +16,15 @@
 
 namespace spaces {
 
-dist_func_t<float> L2_FP32_GetDistFunc(size_t dim, const Arch_Optimization arch_opt) {
+dist_func_t<float> L2_FP32_GetDistFunc(size_t dim, const Arch_Optimization arch_opt,
+                                       unsigned char *alignment) {
+    unsigned char dummy_alignment;
+    if (!alignment) {
+        alignment = &dummy_alignment;
+    }
 
     dist_func_t<float> ret_dist_func = FP32_L2Sqr;
+    *alignment = 4; // Default alignment for float
```

Comment:
Why do we need a default value if we decide not to align?

Reply:
we currently must give some alignment (at least `1`) to use the `DataBlock` constructor without garbage in `alignment`.